src/deploy.sh

```diff
@@ -821,49 +821,103 @@ function modules_install_to()
   cmd_manager "$flag" "$cmd"
 }
 
-# This function handles the config file copy from the host machine to the
-# target.
+# This function is responsible for putting all the required boot files in a
+# single place (~/.cache/kw/to_deploy) to be deployed to the /boot folder
+# later. This function checks if there are dtb/dtbo files; if so, it moves
+# those files.
 #
-# @name Kernel name used in the deploy.
 # @flag How to display a command, the default value is
 #   "SILENT". For more options see `src/kwlib.sh` function `cmd_manager`
 # @target Target can be 1 (VM_TARGET), 2 (LOCAL_TARGET), and 3 (REMOTE_TARGET)
-# @build_and_deploy If the user uses `kw bd` we can safely copy the local
-#                   .config file.
-function deploy_config_file()
+# @kernel_img_name Kernel image file name, e.g., bzImage or Image.
+# @name Kernel name used during the deploy
+# @arch Target device architecture
+function pack_kernel_files_and_send()
 {
-  local name="$1"
-  local flag="$2"
-  local target="$3"
-  local build_and_deploy="$4"
+  local flag="$1"
+  local target="$2"
+  local kernel_img_name="$3"
+  local name="$4"
+  local arch=${5-${configurations[arch]}}
+  local build_and_deploy="$6"
+  local base_boot_path
+  local config_path
   local config_local_version
-  local complain_msg='Undefined .config file for the target kernel. Consider using kw bd'
-  local config_path="$PWD/.config"
-
+  local cache_boot_files_path
+  local compression_type
+  local cache_to_deploy_path
+  local dts_base_path
+  local tar_file_path
+  local kernel_name_arch
+
+  config_path='.config'
+  compression_type="${configurations[deploy_default_compression]}"
+  cache_to_deploy_path="${KW_CACHE_DIR}/${LOCAL_TO_DEPLOY_DIR}"
+  cache_boot_files_path="${cache_to_deploy_path}/boot"
+  tar_file_path="${cache_to_deploy_path}/${name}_boot.tar"
+
+  base_boot_path="arch/$arch_target/boot"
+  dts_base_path="${base_boot_path}/dts"
+
+  # Centralizing kernel files in a single place
+  mkdir -p "$cache_boot_files_path"
+
+  # 1. Copying .config file, we don't want to mislead developers by deploying
+  # the wrong config file.
   if [[ ! -f "$config_path" ]]; then
-    warning "$complain_msg"
-    return
+    warning 'Undefined .config file for the target kernel. Consider using kw bd'
+  else
+    config_local_version=$(sed -nr '/CONFIG_LOCALVERSION=/s/CONFIG_LOCALVERSION="(.*)"/\1/p' "$config_path")
+
+    if [[ -n "$build_and_deploy" || "$name" =~ $config_local_version.*$ ]]; then
+      cmd="cp $config_path ${cache_boot_files_path}/config-$name"
+      cmd_manager "$flag" "$cmd"
+    fi
   fi
 
-  config_local_version=$(sed -nr '/CONFIG_LOCALVERSION=/s/CONFIG_LOCALVERSION="(.*)"/\1/p' "$config_path")
+  # 2. Copy kernel image
+  case "$arch" in
+    'arm')
+      kernel_name_arch="kernel-${name}.img"
```

Comment:
hmmm... I'm not sure if this is kw responsibility. I think this is `.config` change.

Reply:
I mean, the name of the kernel image when you copy it, I would remove the `.img` suffix and use only kernel-${name}. Btw, I tested the last version and the image now is copied as `vmlinuz-{name}` that is weird too :) I guess it is happening because only arch=arm is considered in this switch case, so I think arch=arm64 should be included here too, right?